### PR TITLE
Add docs on skip_validating_missing_parameters in ml-commons connector

### PR DESCRIPTION
**Pull request by yuye-aws**
### Description
Introduce connector parameter in `skip_validating_missing_parameters`.

### Issues Resolved
Introduce connector parameter in `skip_validating_missing_parameters` Closes https://github.com/opensearch-project/documentation-website/issues/8110

### Version
Since OpenSearch 2.17

### Frontend features
_If you're submitting documentation for an OpenSearch Dashboards feature, add a video that shows how a user will interact with the UI step by step. A voiceover is optional._ 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @yuye-aws! I think the table rows got a bit scrambled so I unscrambled them. Please verify that everything looks good now.

**Comment by yuye-aws** [comment]
> Thank you, @yuye-aws! I think the table rows got a bit scrambled so I unscrambled them. Please verify that everything looks good now.

The revision looks good. Just pushed 1 commit to make both parameter description the same. Can you check whether it looks good?

**Comment by kolchfa-aws** [comment]
@yuye-aws Looks good. Applying "Done but waiting to merge" label to put this in the 2.17 editorial queue starting 9/11. This PR will be merged before the release.

File: _ml-commons-plugin/api/connector-apis/update-connector.md [hunk 1]
```diff
@@ -29,17 +29,20 @@ PUT /_plugins/_ml/connectors/<connector_id>
...
+| `description` | String      | A description of the connector.                                                                                                                                                                                                                                                                                                                                                               |
+| `version` | Integer     | The version of the connector.                                                                                                                                                                                                                                                                                                                                                                 |
```
Comment (natebower) [inline _ml-commons-plugin/api/connector-apis/update-connector.md:36]:
```suggestion
| `version` | Integer     | The connector version.                                                                                                                                                                                                                                                                                                                                                                 |
```

File: _ml-commons-plugin/api/connector-apis/update-connector.md [hunk 2]
```diff
@@ -29,17 +29,20 @@ PUT /_plugins/_ml/connectors/<connector_id>
...
-| `backend_roles` | JSON array | A list of OpenSearch backend roles. For more information about setting up backend roles, see [Assigning backend roles to users]({{site.url}}{{site.baseurl}}/ml-commons-plugin/model-access-control#assigning-backend-roles-to-users). |
-| `access_mode` | String | Sets the access mode for the model, either `public`, `restricted`, or `private`. Default is `private`. For more information about `access_mode`, see [Model groups]({{site.url}}{{site.baseurl}}/ml-commons-plugin/model-access-control#model-groups). |
+| `actions` | JSON array  | Defines which actions can run within the connector. If you're an administrator creating a connection, add the [blueprint]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/blueprints/) for your desired connection.                                                                                                                                                              |
```
Comment (natebower) [inline _ml-commons-plugin/api/connector-apis/update-connector.md:40]: Is "creating a connection" necessary here?

File: _ml-commons-plugin/api/connector-apis/update-connector.md [hunk 3]
```diff
@@ -29,17 +29,20 @@ PUT /_plugins/_ml/connectors/<connector_id>
...
+| `access_mode` | String      | Sets the access mode for the model, either `public`, `restricted`, or `private`. Default is `private`. For more information about `access_mode`, see [Model groups]({{site.url}}{{site.baseurl}}/ml-commons-plugin/model-access-control#model-groups).                                                                                                                                        |
+| `parameters.skip_validating_missing_parameters`  | Boolean     | When set to true, this option allows to send a request using a connector without validating any missing parameters. Default is `false`.                                                                                                                                                                                                                                                                     |
```
Comment (natebower) [inline _ml-commons-plugin/api/connector-apis/update-connector.md:43]:
```suggestion
| `parameters.skip_validating_missing_parameters`  | Boolean     | When set to `true`, this option allows you to send a request using a connector without validating any missing parameters. Default is `false`.                                                                                                                                                                                                                                                                     |
```

File: _ml-commons-plugin/remote-models/blueprints.md [hunk 4]
```diff
@@ -55,19 +55,20 @@ As an ML developer, you can build connector blueprints for other platforms. Usin
...
+| `description`                                    | String | Yes         | A description of the connector.                                                                                                                                                                                                                                                                                                                                                      |
+| `version`                                        | Integer | Yes         | The version of the connector.                                                                                                                                                                                                                                                                                                                                                        |
```
Comment (natebower) [inline _ml-commons-plugin/remote-models/blueprints.md:62]:
```suggestion
| `version`                                        | Integer | Yes         | The connector version.                                                                                                                                                                                                                                                                                                                                                        |
```

File: _ml-commons-plugin/remote-models/blueprints.md [hunk 5]
```diff
@@ -55,19 +55,20 @@ As an ML developer, you can build connector blueprints for other platforms. Usin
...
+| `protocol`                                       | String | Yes         | The protocol for the connection. For AWS services such as Amazon SageMaker and Amazon Bedrock, use `aws_sigv4`. For all other services, use `http`.                                                                                                                                                                                                                                  |
```
Comment (natebower) [inline _ml-commons-plugin/remote-models/blueprints.md:63]:
```suggestion
| `protocol`                                       | String | Yes         | The protocol for the connection. For AWS services, such as Amazon SageMaker and Amazon Bedrock, use `aws_sigv4`. For all other services, use `http`.                                                                                                                                                                                                                                  |
```

File: _ml-commons-plugin/remote-models/blueprints.md [hunk 6]
```diff
@@ -55,19 +55,20 @@ As an ML developer, you can build connector blueprints for other platforms. Usin
...
+| `parameters`                                     | JSON object | Yes         | The default connector parameters, including `endpoint`, `model` and `skip_validating_missing_parameters`. Any parameters indicated in this field can be overridden by parameters specified in a predict request.                                                                                                                                                                     |
```
Comment (natebower) [inline _ml-commons-plugin/remote-models/blueprints.md:64]:
```suggestion
| `parameters`                                     | JSON object | Yes         | The default connector parameters, including `endpoint`, `model`, and `skip_validating_missing_parameters`. Any parameters indicated in this field can be overridden by parameters specified in a predict request.                                                                                                                                                                     |
```

File: _ml-commons-plugin/remote-models/blueprints.md [hunk 7]
```diff
@@ -55,19 +55,20 @@ As an ML developer, you can build connector blueprints for other platforms. Usin
...
+| `credential`                                     | JSON object | Yes         | Defines any credential variables required to connect to your chosen endpoint. ML Commons uses **AES/GCM/NoPadding** symmetric encryption to encrypt your credentials. When the connection to the cluster first starts, OpenSearch creates a random 32-byte encryption key that persists in OpenSearch's system index. Therefore, you do not need to manually set the encryption key. |
```
Comment (natebower) [inline _ml-commons-plugin/remote-models/blueprints.md:65]:
```suggestion
| `credential`                                     | JSON object | Yes         | Defines any credential variables required for connecting to your chosen endpoint. ML Commons uses **AES/GCM/NoPadding** symmetric encryption to encrypt your credentials. When the cluster connection is initiated, OpenSearch creates a random 32-byte encryption key that persists in OpenSearch's system index. Therefore, you do not need to manually set the encryption key. |
```

File: _ml-commons-plugin/remote-models/blueprints.md [hunk 8]
```diff
@@ -55,19 +55,20 @@ As an ML developer, you can build connector blueprints for other platforms. Usin
...
+| `actions`                                        | JSON array  | Yes         | Defines what actions can run within the connector. If you're an administrator creating a connection, add the [blueprint]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/blueprints/) for your desired connection.                                                                                                                                                      |
```
Comment (natebower) [inline _ml-commons-plugin/remote-models/blueprints.md:66]: Either "Defines which actions" or "Defines the actions that can". Is "creating a connection" necessary here?

File: _ml-commons-plugin/remote-models/blueprints.md [hunk 9]
```diff
@@ -55,19 +55,20 @@ As an ML developer, you can build connector blueprints for other platforms. Usin
...
+| `client_config`                                  | JSON object | No          | The client configuration object, which provides settings that control the behavior of the client connections used by the connector. These settings allow you to manage connection limits and timeouts, ensuring efficient and reliable communication.                                                                                                                                |
+| `parameters.skip_validating_missing_parameters`  | Boolean | No          | When set to true, this option allows to send a request using a connector without validating any missing parameters. Default is `false`.                                                                                                                                                                                                                                                            |
```
Comment (natebower) [inline _ml-commons-plugin/remote-models/blueprints.md:71]:
```suggestion
| `parameters.skip_validating_missing_parameters`  | Boolean | No          | When set to `true`, this option allows you to send a request using a connector without validating any missing parameters. Default is `false`.                                                                                                                                                                                                                                                            |
```

File: _ml-commons-plugin/remote-models/blueprints.md [hunk 10]
```diff
@@ -76,12 +77,11 @@ The `actions` parameter supports the following options.
 |:---|:---|:---|
 | `action_type` | String | Required. Sets the ML Commons API operation to use upon connection. As of OpenSearch 2.9, only `predict` is supported. |
 | `method`  | String | Required. Defines the HTTP method for the API call. Supports `POST` and `GET`. |
-| `url` | String      | Required. Sets the connection endpoint at which the action occurs. This must match the regex expression for the connection used when [adding trusted endpoints]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/index#adding-trusted-endpoints). |
-| `headers` | JSON object | Sets the headers used inside the request or response body. Default is `ContentType: application/json`. If your third-party ML tool requires access control, define the required `credential` parameters in the `headers` parameter. |
+| `url` | String      | Required. Sets the connection endpoint at which the action occurs. This must match the regex expression for the connection used when [adding trusted endpoints]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/index#adding-trusted-endpoints)|
```
Comment (natebower) [inline _ml-commons-plugin/remote-models/blueprints.md:80]:
```suggestion
| `url` | String      | Required. Specifies the connection endpoint at which the action occurs. This must match the regex expression for the connection used when [adding trusted endpoints]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/index#adding-trusted-endpoints).|
```

File: _ml-commons-plugin/remote-models/blueprints.md [hunk 11]
```diff
@@ -76,12 +77,11 @@ The `actions` parameter supports the following options.
...
 | `post_process_function` | String   | Optional. A built-in or custom Painless script used to post-process the model output data. OpenSearch provides the following built-in post-process functions that you can call directly:<br> - `connector.pre_process.cohere.embedding` for [Cohere text embedding models](https://docs.cohere.com/reference/embed)<br> - `connector.pre_process.openai.embedding` for [OpenAI text embedding models](https://platform.openai.com/docs/api-reference/embeddings) <br> - `connector.post_process.default.embedding`, which you can use to post-process documents in the model response so that they are in the format that neural search expects (OpenSearch 2.11 or later). For more information, see [Built-in functions](#built-in-pre--and-post-processing-functions). |
-
+| `headers` | JSON object | Sets the headers used in the request or response body. Default is `ContentType: application/json`. If your third-party ML tool requires access control, define the required `credential` parameters in the `headers` parameter. |
```
Comment (natebower) [inline _ml-commons-plugin/remote-models/blueprints.md:84]: "Sets" => "Defines" or "Specifies"? Define ML on first appearance if necessary.

**Review by natebower — COMMENTED** [review]
@yuye-aws @kolchfa-aws Please see my comments and changes and let me know if you have any questions. Thanks!

File: _ml-commons-plugin/api/connector-apis/update-connector.md (hunk 2 above)
Comment (kolchfa-aws) [reply _ml-commons-plugin/api/connector-apis/update-connector.md:40]: I think it clarifies the action you're taking.

File: _ml-commons-plugin/remote-models/blueprints.md (hunk 8 above)
Comment (kolchfa-aws) [inline _ml-commons-plugin/remote-models/blueprints.md:66]:
```suggestion
| `actions`                                        | JSON array  | Yes         | Defines the actions that can run within the connector. If you're an administrator creating a connection, add the [blueprint]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/blueprints/) for your desired connection.                                                                                                                                                      |
```

File: _ml-commons-plugin/remote-models/blueprints.md (hunk 11 above)
Comment (kolchfa-aws) [inline _ml-commons-plugin/remote-models/blueprints.md:84]:
```suggestion
| `headers` | JSON object | Specifies the headers used in the request or response body. Default is `ContentType: application/json`. If your third-party ML tool requires access control, define the required `credential` parameters in the `headers` parameter. |
```

Comment (kolchfa-aws) [reply _ml-commons-plugin/remote-models/blueprints.md]: Already defined :)

**Pull request merged**